governator-core/src/main/java/com/netflix/governator/ScanningModuleBuilder.java

```diff
@@ -0,0 +1,231 @@
+     */
+    public ScanningModuleBuilder forPackages(Collection<String> packages) {
+        this.packages = new ArrayList<>(packages);
+        return this;
+    }
+    
+    /**
+     * Specify an {@link AnnotatedClassScanner} to process classes from the set of specified packages.  
+     * 
+     * @param packages
+     * @return Builder for chaining
+     */
+    public ScanningModuleBuilder scanForAnnotatedClasses(AnnotatedClassScanner scanner) {
+        scanners.add(scanner);
+        return this;
+    }
+    
+    /**
+     * Exclude bindings for any class matching the specified predicate.
+     * @param classes
+     * @return Builder for chaining
+     */
+    public ScanningModuleBuilder excludeClasses(Predicate<String> predicate) {
+        excludeRule = Predicates.or(excludeRule, predicate);
+        return this;
+    }
+
+    /**
+     * Exclude specific classes from the classpath scanning.
+     * @param classes
+     * @return Builder for chaining
+     */
+    public ScanningModuleBuilder excludeClasses(String... classes) {
+        return excludeClasses(Arrays.asList(classes));
+    }
+    
+    /**
+     * Exclude specific classes from the classpath scanning.
+     * @param classes
+     * @return Builder for chaining
+     */
+    public ScanningModuleBuilder excludeClasses(Collection<String> classes) {
+        final Collection<String> toTest = new HashSet<>(classes);
+        return excludeClasses(new Predicate<String>() {
+           @Override
+            public boolean apply(String className) {
+               return toTest.contains(className);
+            }
+        });
+    }
+    
+    public Module build() {
+        final Predicate<String> includeRule = Predicates.not(excludeRule);
+        // Generate the list of elements here and immediately create a module from them.  This ensures
+        // that the class path is canned only once as a Module's configure method may be called multiple
+        // times by Guice.
+        return Elements.getModule(Elements.getElements(new AbstractModule() {
+            @Override
+            public void configure() {
+                for ( String basePackage : packages )  {
+                    try {
+                        String basePackageWithSlashes = basePackage.replace(".", "/");
+                        for (URL url : Collections.list(classLoader.getResources(basePackageWithSlashes))) {
+                            try {
+                                if ( isJarURL(url)) {
+                                    String jarPath = url.getFile();
+                                    if ( jarPath.contains("!") ) {
+                                        jarPath = jarPath.substring(0, jarPath.indexOf("!"));
+                                        url = new URL(jarPath);
+                                    }
+                                    File file = ClasspathUrlDecoder.toFile(url);
+                                    try (JarFile jar = new JarFile(file)) {
```

Comment:
better to consolidate all the urls pertaining to a single jar, open it once and read all the jar entries in one shot.